packages/token-bridge-sdk/src/hooks/useArbTokenBridge.ts

```diff
@@ -416,23 +416,43 @@ export const useArbTokenBridge = (
                 'destBridgeAddress' in e
             )
         }
-        if (!isExtensions(extensions))
-          throw new Error('Object not of BridgeInfo format')
-        return extensions.bridgeInfo
+        if (!isExtensions(extensions)){
+          return null
+        } else {
+          return extensions.bridgeInfo
+        }
       })()
 
-      const l1Address = bridgeInfo[await l1NetworkIDCached()].tokenAddress
-
-      bridgeTokensToAdd[l1Address] = {
-        name,
-        type: TokenType.ERC20,
-        symbol,
-        allowed: false,
-        address: l1Address,
-        l2Address,
-        decimals,
-        logoURI,
-        listID
+      if(bridgeInfo){
+        const l1Address = bridgeInfo[await l1NetworkIDCached()].tokenAddress
+
+        bridgeTokensToAdd[l1Address] = {
+          name,
+          type: TokenType.ERC20,
+          symbol,
+          allowed: false,
+          address: l1Address,
+          l2Address: address,
+          decimals,
+          logoURI,
+          listID
+        }
+      }
+      // unbridged L1 token:
+      // stopgap: giant lists (i.e., CMC list) currently severaly hurts page performace, so for now we only add the bridged tokens
```

Comment:
hm it might be slow because of the parsing we doing through the list ie trying to verify an invalid token we go one by one in a list when we could do a binary search style doing it in halfs.
do we know what exactly makes it hang? ie the request or a particular part of the token bridge sdk

Reply:
It's something at the UI/react level; verification is fast, and the ajax request is slow but not the all that bad / not the core issue. Haven't looked deeper yet